python/ray/data/dataset.py

```diff
@@ -1554,7 +1557,7 @@ def train_test_split(
             return ds.split_at_indices([ds_length - test_size])
 
     @ConsumptionAPI(pattern="Args:")
-    def union(self, *other: List["Dataset"]) -> "Dataset":
+    def union(self, *other: List["Dataset"], preserve_order: bool = False) -> "Dataset":
```

Comment:
Preserve order should come from the ExecutionOptions passed to the operator start, not in the API.